core/backend/src/IModelDb.ts

```diff
@@ -223,17 +226,17 @@ const withBriefcaseDb = async (briefcase: OpenBriefcaseArgs, fn: (_db: Briefcase
 
 /**
  * Settings for an individual iModel. May only include settings priority for iModel, iTwin and organization.
- * @note if there is more than one iModel for an iTwin or organization, they will *each* hold a copy of the settings for those priorities.
+ * @note if there is more than one iModel for an iTwin or organization, they will *each* hold an independent copy of the settings for those priorities.
  */
 class IModelSettings extends BaseSettings {
-  protected override verifyPriority(priority: SettingsPriority) {
-    if (priority <= SettingsPriority.application)
+  protected override verifyPriority(priority: Settings.Priority) {
+    if (priority <= Settings.Priority.application)
       throw new Error("Use IModelHost.appSettings");
   }
 
-  // attempt to resolve a setting from this iModel's settings, otherwise use appWorkspace's settings, otherwise defaultValue.
-  public override resolveSetting<T extends SettingType>(name: SettingName, resolver: SettingResolver<T>, defaultValue?: T): T | undefined {
-    return super.resolveSetting(name, resolver) ?? IModelHost.appWorkspace.settings.resolveSetting(name, resolver, defaultValue);
+  // attempt to resolve a setting from this iModel's settings, otherwise optionally use appWorkspace's settings, otherwise defaultValue.
+  public override resolveSetting<T extends SettingType>(arg: { settingName: SettingName, resolver: Settings.Resolver<T>, skipAppWorkspace?: boolean }, defaultValue?: T): T | undefined {
```

Comment:
I don't see any way that a user of `IModelDb.workspace.settings` could pass the `skipAppWorkspace` flag, which exist only here on this non-exported type. Is it useful and if so does it belong in the base interface as something like `noCascade`?